### ISSUE TITLE
CompressionSpec failure

### ISSUE THREAD

**Issue by alissapajer** [quasar-analytics/quasar#2208]
```
CompressionSpec
[info] coalesceWithUnknown
[error] x merges known map entry with unknown entry when same primary type appears in unknown (1 second, 714 ms)
[error] Falsified after 91 passed tests.
[error] > ARG_0: NonEmpty[(䍩,LeafEjs(Fix(Coproduct(\/-(Bool(true)))))),(ᗴ,LeafEjs(Fix(Coproduct(\/-(Bool(false)))))),(㔄,LeafEjs(Fix(Coproduct(\/-(Str()))))),(〳,LeafEjs(Fix(Coproduct(-\/(Int(785206339)))))),(恿,LeafEjs(Fix(Coproduct(\/-(Null()))))),(﫯,LeafEjs(Fix(Coproduct(\/-(Bool(false)))))),(﫯,LeafEjs(Fix(Coproduct(-\/(Int(2147483647)))))),(僩,LeafEjs(Fix(Coproduct(\/-(Str(蕡쐕昹丬姙))))))]
[error] > ARG_1: (0,LeafEjs(Fix(Coproduct(-\/(Int(-1723640382))))))
[error] > ARG_1_ORIGINAL: (-1993951737,LeafEjs(Fix(Coproduct(-\/(Int(-1723640382))))))
[error] > StructuralType(Some(Count(1)), {`Int(0)` : (Some(Int(SampleStats["n = 1","μ = -1723640382","σ² = ?","σ = ?"], -1723640382, -1723640382)), `Int(-1723640382)`) ? (Some(Char(8, x, 﫯)), ((Some(Char(1, x, x)), `Char(x)`) | (Some(Char(1, 﫯, 﫯)), `Char(﫯)`) | (Some(Char(1, 恿, 恿)), `Char(恿)`) | (Some(Char(1, 僩, 僩)), `Char(僩)`) | (Some(Char(1, 䍩, 䍩)), `Char(䍩)`) | (Some(Char(1, 㔄, 㔄)), `Char(㔄)`) | (Some(Char(1, ᗴ, ᗴ)), `Char(ᗴ)`) | (Some(Char(1, 〳, 〳)), `Char(〳)`))) : (Some(Count(8)), ((Some(Bool(2, 0)), `Bool(true)`) | (Some(Str(1, 0, 0, "", "")), `Str()`) | (Some(Bool(0, 1)), `Bool(false)`) | (Some(Int(SampleStats["n = 1","μ = 785206339","σ² = ?","σ = ?"], 785206339, 785206339)), `Int(785206339)`) | (Some(Int(SampleStats["n = 1","μ = 2147483647","σ² = ?","σ = ?"], 2147483647, 2147483647)), `Int(2147483647)`) | (Some(Count(1)), `Null()`) | (Some(Str(1, 5, 5, "蕡쐕昹丬姙", "蕡쐕昹丬姙")), `Str(蕡쐕昹丬姙)`)))}) !== StructuralType(Some(Count(1)), {`Int(0)` : (Some(Int(SampleStats["n = 1","μ = -1723640382","σ² = ?","σ = ?"], -1723640382, -1723640382)), `Int(-1723640382)`) ? (Some(Char(9, x, 﫯)), ((Some(Char(1, x, x)), `Char(x)`) | (Some(Char(1, 僩, 僩)), `Char(僩)`) | (Some(Char(2, 﫯, 﫯)), `Char(﫯)`) | (Some(Char(1, 恿, 恿)), `Char(恿)`) | (Some(Char(1, 〳, 〳)), `Char(〳)`) | (Some(Char(1, 㔄, 㔄)), `Char(㔄)`) | (Some(Char(1, 䍩, 䍩)), `Char(䍩)`) | (Some(Char(1, ᗴ, ᗴ)), `Char(ᗴ)`))) : (Some(Count(9)), ((Some(Bool(2, 0)), `Bool(true)`) | (Some(Str(1, 5, 5, "蕡쐕昹丬姙", "蕡쐕昹丬姙")), `Str(蕡쐕昹丬姙)`) | (Some(Int(SampleStats["n = 1","μ = 2147483647","σ² = ?","σ = ?"], 2147483647, 2147483647)), `Int(2147483647)`) | (Some(Bool(0, 2)), `Bool(false)`) | (Some(Count(1)), `Null()`) | (Some(Int(SampleStats["n = 1","μ = 785206339","σ² = ?","σ = ?"], 785206339, 785206339)), `Int(785206339)`) | (Some(Str(1, 0, 0, "", "")), `Str()`)))}) (QuasarSpecification.scala:51)
```

**Comment by alissapajer** [comment]
Second example:
```
[info] coalesceWithUnknown
[error] x merges known map entry with unknown entry when same primary type appears in unknown (992 ms)
[error]  Falsified after 80 passed tests.
[error]  > ARG_0: NonEmpty[(❑,LeafEjs(Fix(Coproduct(\/-(Bool(false)))))),(ꞃ,LeafEjs(Fix(Coproduct(\/-(Null()))))),(,LeafEjs(Fix(Coproduct(\/-(Dec(-0.02413459454118724125833285815869110)))))),(젬,LeafEjs(Fix(Coproduct(-\/(Char(뾲)))))),(ꅚ,LeafEjs(Fix(Coproduct(-\/(Int(1656131619)))))),(젬,LeafEjs(Fix(Coproduct(\/-(Dec(9.223372036854775808E+104)))))),(⬊,LeafEjs(Fix(Coproduct(\/-(Str(訫墬꿣顩쒠))))))]
[error]  > ARG_1: (0,LeafEjs(Fix(Coproduct(\/-(Null())))))
[error]  > ARG_1_ORIGINAL: (619054408,LeafEjs(Fix(Coproduct(\/-(Null())))))
[error]  > StructuralType(Some(Count(1)), {`Int(0)` : (Some(Count(1)), `Null()`) ? (Some(Char(7, x, )), ((Some(Char(1, x, x)), `Char(x)`) | (Some(Char(1, , )), `Char()`) | (Some(Char(1, 젬, 젬)), `Char(젬)`) | (Some(Char(1, ꞃ, ꞃ)), `Char(ꞃ)`) | (Some(Char(1, ꅚ, ꅚ)), `Char(ꅚ)`) | (Some(Char(1, ❑, ❑)), `Char(❑)`) | (Some(Char(1, ⬊, ⬊)), `Char(⬊)`))) : (Some(Count(7)), ((Some(Bool(0, 2)), `Bool(false)`) | (Some(Str(1, 5, 5, "訫墬꿣顩쒠", "訫墬꿣顩쒠")), `Str(訫墬꿣顩쒠)`) | (Some(Dec(SampleStats["n = 1","μ = -241345945411872412583328581586911/10000000000000000000000000000000000","σ² = ?","σ = ?"], -0.02413459454118724125833285815869110, -0.02413459454118724125833285815869110)), `Dec(-0.02413459454118724125833285815869110)`) | (Some(Dec(SampleStats["n = 1","μ = 922337203685477580800000000000000000000000000000000000000000000000000000000000000000000000000000000000000","σ² = ?","σ = ?"], 9.223372036854775808E+104, 9.223372036854775808E+104)), `Dec(9.223372036854775808E+104)`) | (Some(Count(1)), `Null()`) | (Some(Int(SampleStats["n = 1","μ = 1656131619","σ² = ?","σ = ?"], 1656131619, 1656131619)), `Int(1656131619)`)))}) !== StructuralType(Some(Count(1)), {`Int(0)` : (Some(Count(1)), `Null()`) ? (Some(Char(8, x, )), ((Some(Char(1, x, x)), `Char(x)`) | (Some(Char(1, ⬊, ⬊)), `Char(⬊)`) | (Some(Char(2, 젬, 젬)), `Char(젬)`) | (Some(Char(1, , )), `Char()`) | (Some(Char(1, ❑, ❑)), `Char(❑)`) | (Some(Char(1, ꞃ, ꞃ)), `Char(ꞃ)`) | (Some(Char(1, ꅚ, ꅚ)), `Char(ꅚ)`))) : (Some(Count(8)), ((Some(Bool(0, 2)), `Bool(false)`) | (Some(Count(1)), `Null()`) | (Some(Str(1, 5, 5, "訫墬꿣顩쒠", "訫墬꿣顩쒠")), `Str(訫墬꿣顩쒠)`) | (Some(Dec(SampleStats["n = 1","μ = 922337203685477580800000000000000000000000000000000000000000000000000000000000000000000000000000000000000","σ² = ?","σ = ?"], 9.223372036854775808E+104, 9.223372036854775808E+104)), `Dec(9.223372036854775808E+104)`) | (Some(Int(SampleStats["n = 1","μ = 1656131619","σ² = ?","σ = ?"], 1656131619, 1656131619)), `Int(1656131619)`) | (Some(Char(1, 뾲, 뾲)), `Char(뾲)`) | (Some(Dec(SampleStats["n = 1","μ = -241345945411872412583328581586911/10000000000000000000000000000000000","σ² = ?","σ = ?"], -0.02413459454118724125833285815869110, -0.02413459454118724125833285815869110)), `Dec(-0.02413459454118724125833285815869110)`)))}) (QuasarSpecification.scala:51)
```

**Comment by alissapajer** [comment]
Different test but maybe the same problem:
```
[error] x merges known map entry with unknown when primary type appears in unknown union (1 second, 289 ms)
[error]  Falsified after 70 passed tests.
[error]  > ARG_0: NonEmpty[(,LeafEjs(Fix(Coproduct(-\/(Byte(0)))))),(筇,LeafEjs(Fix(Coproduct(-\/(Byte(18)))))),(鎔,LeafEjs(Fix(Coproduct(-\/(Char(弓)))))),(鎔,LeafEjs(Fix(Coproduct(\/-(Bool(false)))))),(⊡,LeafEjs(Fix(Coproduct(-\/(Byte(0)))))),(삷,LeafEjs(Fix(Coproduct(-\/(Int(255859705))))))]
[error]  > ARG_1: (0,LeafEjs(Fix(Coproduct(-\/(Byte(1))))))
[error]  > ARG_1_ORIGINAL: (-416325339,LeafEjs(Fix(Coproduct(-\/(Byte(1))))))
[error]  > StructuralType(Some(Count(1)), {`Int(0)` : (Some(Byte(1, 1, 1)), `Byte(1)`) ? (Some(Count(6)), ((Some(Count(1)), ⊤) | (Some(Char(1, , )), `Char()`) | (Some(Char(1, 삷, 삷)), `Char(삷)`) | (Some(Char(1, 鎔, 鎔)), `Char(鎔)`) | (Some(Char(1, ⊡, ⊡)), `Char(⊡)`) | (Some(Char(1, 筇, 筇)), `Char(筇)`) | (Some(Char(1, x, x)), `Char(x)`) | (Some(Count(1)), decimal))) : (Some(Count(6)), ((Some(Byte(3, 0, 0)), `Byte(0)`) | (Some(Byte(1, 18, 18)), `Byte(18)`) | (Some(Int(SampleStats["n = 1","μ = 255859705","σ² = ?","σ = ?"], 255859705, 255859705)), `Int(255859705)`) | (Some(Bool(0, 1)), `Bool(false)`)))}) !== StructuralType(Some(Count(1)), {`Int(0)` : (Some(Byte(1, 1, 1)), `Byte(1)`) ? (Some(Count(7)), ((Some(Count(1)), ⊤) | (Some(Char(1, 삷, 삷)), `Char(삷)`) | (Some(Char(1, ⊡, ⊡)), `Char(⊡)`) | (Some(Char(2, 鎔, 鎔)), `Char(鎔)`) | (Some(Char(1, , )), `Char()`) | (Some(Char(1, 筇, 筇)), `Char(筇)`) | (Some(Char(1, x, x)), `Char(x)`) | (Some(Count(1)), decimal))) : (Some(Count(7)), ((Some(Byte(3, 0, 0)), `Byte(0)`) | (Some(Byte(1, 18, 18)), `Byte(18)`) | (Some(Bool(0, 1)), `Bool(false)`) | (Some(Char(1, 弓, 弓)), `Char(弓)`) | (Some(Int(SampleStats["n = 1","μ = 255859705","σ² = ?","σ = ?"], 255859705, 255859705)), `Int(255859705)`)))}) (QuasarSpecification.scala:51)
```
